pkg/staticnetworkconfig/generator.go

```diff
@@ -183,11 +183,21 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 		if err != nil {
 			return errors.Wrapf(err, "failed to get interface type for interface %s", interfaceName.String())
 		}
-		if lo.Contains([]string{"802-3-ethernet", "ethernet"}, interfaceType.String()) {
+		switch interfaceType.String() {
+		case "802-3-ethernet", "ethernet":
 			if !lo.Contains(interfaceNames, interfaceName.String()) {
 				return errors.Errorf("mac-interface mapping for interface %s is missing", interfaceName.String())
 			}
 			matched = true
+		case "vlan":
+			vlanSection := cfg.Section("vlan")
+			parent, err := vlanSection.GetKey("parent")
+			if err != nil {
+				return errors.Wrapf(err, "failed to get parent of vlan %s", interfaceName.Name())
+			}
+			if lo.Contains(interfaceNames, parent.String()) {
```

Comment:
With bond, slave connection files are created by nmstate - so they should be matched as ethernet connection files.  Besides, parent reference in bond is by UUID and not interface name.
Bridge interfaces are not supported.

Reply:
I guess that answers my previous comment.
But as we discussed offline, just summarizing - since this nmstate config is the initial config of the node, if there is a vlan on top of a bond, the bond should have it's connection file as well.

Reply:
As mentioned previously, this is not the purpose of this PR.

Reply:
Maybe my comment wasn't clear. I just summarized what you explained me offline to keep history on the PR.
I asked in the first comment what if the parent is bond/bridge and the answer is that since it is the first boot, only the physical interfaces pre exist on the node, so even if the parent of a vlan is a bond - the bond will have its own connection file as well (since for sure it is not defined on the node yet).